docs: Basic cookbook template using quarto

rodriguesfred opened the pull request:
## Description

### Summary
Created a basic folder structure for a Quarto cookbook.
Created an action to render and deploy cookbook to gihub pages

## Type of Change
<!-- Mark the relevant option with an "x" -->
- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📚 Documentation update
- [ ] 🔧 Refactoring (no functional changes, no api changes)
- [ ] ⚡ Performance improvement
- [x] 🧪 Test addition or improvement
- [ ] 🔨 Build/CI changes
- [ ] 🎨 Style/formatting changes

## Changes Made
Created a basic folder structure for a Quarto cookbook.
Created an action to render and deploy cookbook to gihub pages

## Testing
Confirmed the cookbook deploys to github pages on [https://best-practice-and-impact.github.io/afcharts-py/]

## Documentation
- [ ] Documentation updated
- [x] README updated
- [ ] API documentation updated
- [ ] No documentation changes needed

**Documentation Changes:**
Added a section onto the readme on how to test the cookbook

## Dependencies
- [ ] No new dependencies
- [x] New dependencies added (list below)
- [ ] Dependencies updated (list below)
- [ ] Dependencies removed (list below)

**Dependency Changes:**
Requires Quarto installed locally to test the cookbook.

## Checklist
<!-- Mark completed items with an "x" -->

### Code Quality
- [ ] Code follows the project's style guidelines
- [ ] Self-review of code completed
- [ ] Code is properly commented
- [ ] No unnecessary console.log/print statements
- [ ] No commented-out code (unless intentionally documented)

### Testing & Quality Assurance
- [ ] Tests pass locally
- [ ] Tests added for new functionality
- [ ] Edge cases considered and tested
- [ ] Error handling implemented where appropriate







**Github issue link**:  #_\<Enter GitHub issue ref here\>_

## Describe your changes

_\<Enter a description for this PR here\>_

## Checklist before requesting a review

The PR raiser/assignee must ensure that for:
- All PRs:
    - [ ] The [**branching strategy**] (branching of the `main` branch) has been followed.
    - [ ] A **self-review** of the changes has been performed by the author.
    - [ ] Any **merge conflicts** have been resolved (see bottom of PR).
    -  All PR **workflow checks** have passed (see outcomes at bottom of PR):
        - [ ] The [**Conventional Commit**](https://www.conventionalcommits.org/en/v1.0.0) messages format has been followed for the PR title (see README.md).
        - [ ] All **linting issues** raised by MegaLinter have been resolved.
        - [ ] All **tests** have passed.
- New features or bugfixes (less relevant for `doc`,`style`, `chore` etc.):
    - [ ] Thorough **tests** have been added/updated for modified code.
    - [ ] Updates to **documentation** have been made as necessary.

rodriguesfred commented:
Could someone review? So we can close this PR and #26

nickhenden commented on src/afcharts/cookbook/getting-started.qmd:
````diff
@@ -0,0 +1,35 @@
+pip install 'afcharts_py'
+```
+
+**use_afcharts** <br>
+The examples in this cookbook use the afcharts theme and colour functions explicitly, however it may be easier to make use of the `use_afcharts()` function if your charts all require a similar style. More information on `use_afcharts` can be found on the [homepage](https://best-practice-and-impact.github.io/afcharts/#use-afcharts-as-default).
````
Remove for now. Not sure if we'll have a function like this or what the syntax will be

nickhenden commented on src/afcharts/cookbook/getting-started.qmd:
```diff
@@ -0,0 +1,35 @@
+**Note on use of titles, subtitles and captions** <br>
+Titles, subtitles and captions have been embedded in the charts in this cookbook for demonstration purposes. However, for accessibility reasons, it is usually preferable to provide titles in the body of the page rather than embedded within the image of the plot.
+
+The following packages are required to produce the example charts in this cookbook:
```
```suggestion
**Dependencies** <br>
The following packages are required to produce the example charts in this cookbook:
```

nickhenden commented on src/afcharts/cookbook/images/logo.svg:
The logo is already in the root `assets/` folder, can we use that rather than duplicate it here?

rodriguesfred commented on src/afcharts/cookbook/04-colour-palettes.qmd:
````diff
@@ -0,0 +1,98 @@
+The `main` palette should be used with categorical data.
+
+
+```python
+import pandas as pd
+
+test_palette = {
+    "primary": "#1f77b4",
+    "secondary": "#ff7f0e",
+    "success": "#2ca02c",
+    "danger": "#d62728",
+    "warning": "#ffbb78",
+    "info": "#17becf",
+}
+
+df = pd.DataFrame(list(test_palette.items()), columns=["Name", "Hex Code"])
+df
+
+df["Preview"] = df["Hex Code"].apply(
+    lambda x: f'<span style="background-color:{x};padding:5px 20px;display:inline-block;"></span>'
+)
+from IPython.display import display, HTML
+
+display(HTML(df.to_html(escape=False)))
+```
````
its a good point. I think at the moment its written as raw code. Not as a script. One to figure out later but yes Quarto should be able to render.

nickhenden replied:
The charts now render (yay) but this bit of code with the colours doesn't :/

nickhenden commented on .gitignore:
```diff
@@ -133,3 +133,8 @@ README_files/*
 .quarto/
 _book/
+*.html
+*.md
```
Remove `*.md`. Don't want to ignore the README etc.

nickhenden replied:
You can do e.g. `src/afcharts/cookbook/*.md`

nickhenden commented on src/afcharts/cookbook/03-plotly-usage.qmd:
````diff
@@ -41,24 +41,22 @@ fig.update_layout(
 fig.show()
 ```
 
 ##/ Example of line chart with duo palette
````
Why the forward slash? It doesn't seem to render as a heading. Should just be
```suggestion
## Example of a line chart with duo palette
```

nickhenden commented on src/afcharts/cookbook/03-plotly-usage.qmd:
````diff
@@ -101,12 +99,64 @@ fig.update_layout(
 fig.show()
 ```
+
+##/ Example of scatter chart
````
```suggestion
## Example of a scatter chart
```

nickhenden commented on src/afcharts/cookbook/_quarto.yml:
```diff
@@ -5,7 +5,7 @@ project:
   title: "afcharts-py cookbook"
   repo-branch: main
-  favicon: images/logo.svg
+  favicon: ../assets/logo.svg
```
Logo doesn't seem to be rendering now sadly :/ At least, not on https://best-practice-and-impact.github.io/afcharts-py/04-colour-palettes.html

nickhenden commented on src/afcharts/cookbook/index.qmd:
```diff
@@ -5,14 +5,11 @@ description-meta: |
 ::: {.content-visible when-format="html"}
-# Welcome {.unnumbered}
-This cookbook provides examples of the code used to produce various chart types using `afcharts_py`. There are also examples to demonstrate how to apply further customisation to `afcharts_py` charts.
+# Welcome <img src="../assets/logo.svg" alt="afcharts logo" style="float: right; height: 150px;"> {.unnumbered}
```
Logo also doesn't render here

nickhenden commented on tests/plotly_scatter_test.py:
```diff
@@ -0,0 +1,61 @@
+from afcharts.theme_af import theme_af
+
+
+def test_plolty_line_chart():
```
```suggestion
def test_plotly_line_chart():
```

nickhenden reviewed (commented):
Charts look great! Though sadly the logo doesn't render.

rodriguesfred commented:
@nickhenden after several issues deploying that were actually due to github systems were down I've addressed all the comments but struggling with the logo outside the quarto folder.

I got the colours qmd to render the colours
